dvc/repo/reproduce.py

```diff
@@ -125,6 +125,10 @@ def reproduce(  # noqa: C901
                 )
             )
 
+    if kwargs.get("pull", False):
+        logger.debug("Pulling run cache")
+        self.stage_cache.pull(None)
```

Comment:
This might take awhile, but I guess this is acceptable with `--pull`. But still hope to finally get to converting run-cache to dvc-objects to that this operation could be efficient.